gcloud/test_connection.py

```diff
@@ -330,10 +330,35 @@ def test_api_request_w_500(self):
         conn = self._makeMockOne()
         conn._http = _Http(
             {'status': '500', 'content-type': 'text/plain'},
-            '{}',
+            b'{}',
         )
         self.assertRaises(InternalServerError, conn.api_request, 'GET', '/')
 
+    def test_api_request_non_binary_response(self):
+        conn = self._makeMockOne()
+        http = conn._http = _Http(
+            {'status': '200', 'content-type': 'application/json'},
+            u'{}',
```

Comment:
But why are we mocking a non-bytes return?  The point of #724 is that htttplib2 always gives us bytes as the payload, never unicode.